packages/react-dom-bindings/src/client/ReactDOMComponentTree.js

```diff
@@ -112,9 +148,15 @@ export function getClosestInstanceFromNode(targetNode: Node): null | Fiber {
     // itself because the fibers are conceptually between the container
     // node and the first child. It isn't surrounding the container node.
     // If it's not a container, we check if it's an instance.
-    targetInst =
-      (parentNode: any)[internalContainerInstanceKey] ||
-      (parentNode: any)[internalInstanceKey];
+    if (enableInternalInstanceMap) {
+      targetInst =
+        (parentNode: any)[internalContainerInstanceKey] ||
+        internalInstanceMap.get(((parentNode: any): InstanceUnion));
```

Comment:
below you check internalInstanceMap first instead of the node itself, is it intentional to have different precedence in different places?

Reply:
In this version I've moved the main instance handle into the internal map but left the container instance handle on the node. So I'm mirroring the existing checks in terms of precedence. This one checks for container first while the other falls back to container.